src/chainrules.jl

```diff
@@ -2,6 +2,48 @@ ChainRulesCore.@scalar_rule(xlogx(x::Real), (1 + log(x),))
 ChainRulesCore.@scalar_rule(xlogy(x::Real, y::Real), (log(y), x / y,))
 ChainRulesCore.@scalar_rule(xlog1py(x::Real, y::Real), (log1p(y), x / (1 + y),))
 
+function ChainRulesCore.frule((_, Δx), ::typeof(xexpx), x::Real)
+    e = exp(x)
+    if iszero(e)
+        Ω = e
+        ΔΩ = e * Δx
+    else
+        Ω = x * e
+        ΔΩ = (1 + x) * e * Δx
+    end
+    return Ω, ΔΩ
+end
+
+function ChainRulesCore.rrule(::typeof(xexpx), x::Real)
+    e = exp(x)
+    Ω = iszero(e) ? e : x * e
+    xexpx_pullback(Δ) = (ChainRulesCore.NoTangent(), iszero(e) ? e * Δ : (1 + x) * e * Δ)
+    return Ω, xexpx_pullback
+end
+
+function ChainRulesCore.frule((_, Δx, Δy), ::typeof(xexpy), x::T, y::T) where {T<:Real}
+    e = exp(y)
+    if iszero(e) && !isnan(x)
+        Ω = e
+        ΔΩ = e * Δx + zero(Ω * Δy)
+    else
+        Ω = x * e
+        ΔΩ = e * Δx + Ω * Δy
+    end
```

Comment:
`e * Δx + zero(Ω * Δy)` is the same as `e * Δx + Ω * Δy`, isn't it? Similar to above, I'd also suggest using more descriptive variable names, e.g.
```suggestion
    expy = exp(y)
    Ω = iszero(expy) && !isnan(x) ? expy : x * expy
    ΔΩ = expy * Δx + Ω * Δy
```